packages/next-frontend/src/components/step/index.tsx

```diff
@@ -31,44 +36,37 @@ export function Step({
         matchChildByType(child, StepContent),
     ) as ReactElement;
 
+    useEffect(() => {
+        animate({
+            height: (open ? wrapperRef?.current?.offsetHeight : 83) + "px",
+            config: { duration: 200, easing: easings.easeInOutCubic },
+        });
+    }, [animate, open]);
+
     function handlePreviewOnClick() {
         setOpen((open) => !open);
-        setHeight(open ? 0 : "auto");
     }
 
     function handleContentOnClick() {
         if (close === "manual") return;
         setOpen((open) => !open);
-        setHeight(open ? 0 : "auto");
-    }
-
-    // Track the height animation status for the StepPreview component.
-    // This status is used to dynamically apply a CSS class to remove the bottom border rounding
-    // only after the animation has completed.
-    function handleContentAnimationOnEnd() {
-        setHeightAnimationDone((state) => !state);
     }
 
     return (
-        <div
+        <animated.div
+            style={spring}
             className={classNames(styles.root, {
                 [styles.rootDisabled]: disabled,
             })}
         >
-            <div onClick={handlePreviewOnClick}>
-                {React.cloneElement<StepPreviewProps>(previewChildren, {
-                    open,
-                    heightAnimationDone,
-                })}
-            </div>
-            <AnimateHeight
-                height={height}
-                duration={200}
-                easing="ease-in-out"
-                onHeightAnimationEnd={handleContentAnimationOnEnd}
-            >
+            <div ref={wrapperRef}>
+                <div onClick={handlePreviewOnClick}>
+                    {React.cloneElement<StepPreviewProps>(previewChildren, {
```

Comment:
Why do we need the clone here? To pass a prop?

Reply:
Yes, here we need to pass `open` to the `preview` child component, to apply the proper styles.

Reply:
And isn't it possible to do this without cloning by properly typing the `previewChildren` as a function coomponent that takes in that prop?